src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -1222,7 +1222,8 @@ def _build_app(args, extra_args):
 
 def build(args):
     try:
-        executable_id = _build_app(args, process_extra_args(args) if args.extra_args else {})
+        process_extra_args(args)
```

Comment:
`process_extra_args` updates args but not returns it. 
other references:
https://github.com/dnanexus/dx-toolkit/blob/master/src/python/dxpy/scripts/dx.py#L3560
 https://github.com/dnanexus/dx-toolkit/blob/master/src/python/scripts/dx-jobutil-new-job#L73

Reply:
process_extra_args is now called even if args.extra_args does not exist.

Reply:
> process_extra_args

yes in https://github.com/dnanexus/dx-toolkit/blob/a83f740f986d750caa64a02ad759dc34f578f95e/src/python/dxpy/cli/parsers.py#L219
the parsing happens only when extra args are not None @mhrvol